subsys/bluetooth/controller/ll_sw/ull_conn.c

```diff
@@ -1139,9 +1139,9 @@ void ull_conn_tx_lll_enqueue(struct ll_conn *conn, u8_t count)
 
 	tx = conn->tx_head;
 #if defined(CONFIG_BT_CTLR_LE_ENC)
-	while (tx && ((tx == conn->tx_ctrl) || !conn->pause_tx) && count--) {
+	while (tx && (!conn->pause_tx || (tx == conn->tx_ctrl)) && count--) {
```

Comment:
rhs and lhs of logical OR have been swapped. 
I don't see short circuit evaluation affecting the outcome or safety of this.
Style or common-case optimization?

Reply:
Thinking out loud:

1. We are in `ull_conn_tx_lll_enqueue`, so we are pouring `node_tx`'es out from ull and down into lll.
2. Thus `tx` is loop variable, pointing to currently processed `node_tx`.
3. The while condition: we must have a valid current `tx`, and we also cap the loop at `count`. Furthermore, we must obviously not have paused transmission, however, an exception is opened up for LLCP tx. This is because encryption is a whole transactional llcp process. While that process is ongoing, normal data tx must be paused -- however the encryption process must be allowed to continue.

@cvinayak Is that correct?

I guess (if we wanted) we could make it a bit stricter on a specific new queue, `tx_ctrl_crypt` (e.g.) rather than `tx_ctrl`. But IIRC, llcp procedures cannot overlap so the common control queue should be fine.
@mtpr-ot Was it this you mentioned?

Reply:
@mped-oticon The reason to swap, !conn->pause_tx is highly likely case most of the time compared to (tx == conn->tx_ctrl), which will lead to (early) short-circuit of the OR evaluation.

Data and control packets are serialized (must for encryption setup), hence a single list suffices.